### DISCUSSION TITLE
Rules for updating locale files (Mage_[...].csv)

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2590] [Ideas]
Regarding PR #1359, but also to others where there were similar discussions. In that PR, it was proposed to remove the word "Manage" from links/headers in the Backend. I am not discussing whether it is a good or bad proposal for a visual improvement, but the way of implementation. In such situations, the PHP and local CSV files are modified. Let's continue to analyze several situations.

The first in which the modification is in the PHP files. For example, in the file ```/app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute.php``` in line 41 it was proposed:

**Initial**: $this->_headerText = Mage::helper('catalog')->__('Manage Attributes');

**Modified**: $this->_headerText = Mage::helper('catalog')->__('Attributes');

No changes were made in the CSV file because the phrase "Attributes" already exists. As a personal opinion, I agree with such changes, but if the new phrase does not exist in the CSV file, it must be added and announced so that everyone can update their own files.

The second in which the modification is in the CSV files. For example, in the file /app/locale/en_US/Mage_ProductAlert.csv line 26:

**Initial**: "The product is not found.","The product is not found."

**Modified**: "The product was not found.","The product was not found."

The logical error remained in the PHP file, but it was chosen to make the change only in the CSV file. As a personal opinion, the modification should have been made in the PHP file then if the phrase does not exist in the CSV, it should be added. Finally, this change had to be announced so that everyone could update their translation files.

The third in which the modification is in both the PHP and CSV files. Usually this happens when something new like a feature is added. As a personal opinion, the modification of the CSV file should be announced so that everyone can translate those newly introduced phrases.

I do not agree to see different phrases in the same row in the CSV files, it is true that there are about 5 such situations and they should be corrected as soon as possible. Also, a BC issue has been invoked every time, but I believe that it is not a real one because if I make a change in the English CSV file in the second sentence in a row (as OpenMage is now), all those who have customized translations, they will also have to make the changes manually. Therefore, this stage cannot be skipped.

In conclusion, I know that this topic has been discussed, but rules must be established regarding translations when changing PHP and CSV files and most importantly they must be announced.

PS - I don't use translations, but with this post I am the advocate of those who use translations.

**Comment by sreichel** [comment]
Maybe I am wrong ...

Every change to translatable strings in php/xml-files breaks locale packs, that rely on that string. Not?

Removing the word "Manage" has an impact on EVERY, EVERY locale pack. The output in backend changes and i dont know if its wanted in any case. In other languages the wording for e.g. "Manage Producs" an  just "Products" is different ...

IMHO ...

- we should not touch them at all. If we have to change wording or typos, we can do this in CSV files.
- new string should be added to separate file - to make clear that this belongs to OpenMage (again, we already had this)
- incompatible changes (e.g. added variables to string) should be added to separate file
- (revert or at least check all changes since 1.9.4.5)

Btw ... you can also use your custom translation using "inline translations" that are store in DB.
